src/backend/impl_safe/environment.rs

```diff
@@ -152,16 +154,32 @@ impl EnvironmentImpl {
         Ok(bincode::serialize(&data)?)
     }
 
-    fn deserialize(
-        bytes: &[u8],
-        discard_if_corrupted: bool,
+    fn load(
+        path: &Path,
+        strategy: RecoveryStrategy,
     ) -> Result<(DatabaseArena, DatabaseNameMap), ErrorImpl> {
+        let bytes = fs::read(path)?;
+
+        match Self::deserialize(&bytes) {
+            Ok((arena, name_map)) => Ok((arena, name_map)),
+            Err(err) => match strategy {
+                RecoveryStrategy::Error => Err(err),
+                RecoveryStrategy::Discard => Ok((DatabaseArena::new(), HashMap::new())),
+                RecoveryStrategy::Rename => {
+                    let corrupted_path = path.with_extension(DEFAULT_CORRUPT_DB_EXTENSION);
+                    fs::rename(path, corrupted_path)?;
```

Comment:
I guess the assumption was that one deals with the `$db.corrupt` file soon, before hitting another problem. But I can see that that's not always possible.
If we use timestamps rkv-users need to make sure to actually clean up to avoid piling up old corrupted files.

Reply:
By using Rename I'd expect the user to deal with the files, indeed. I wonder what they can really do with it if rkv can't do anything, though.

Reply:
For debugging purporses someone could manually explore that file and see if they are able to detect why the file got corrupted.

I thought about the `.corrupt` vs `.$timestamp` option again and I would say we should stay with `.corrupt` for now:
1. It's what's been discussed a while back
2. It avoids piling up files over files
3. The logic to find it is kept simple
4. We still have the option in the future to change it. Let's see if it's even gonna be adopted inside m-c